superset-frontend/src/explore/components/controls/SelectControl.test.jsx

```diff
@@ -30,139 +37,198 @@ const defaultProps = {
-      wrapper.setProps({ freeForm: false });
-      expect(wrapper.find(SelectComponent)).toExist();
-      expect(wrapper.find(SelectComponent).prop('allowNewOptions')).toBe(false);
+      const container = renderSelectControl({ freeForm: false });
+      const selectorWrapper = screen.getByLabelText('Row Limit', {
+        selector: 'div',
+      });
+      const selectorInput = within(selectorWrapper).getByLabelText(
+        'Row Limit',
+        { selector: 'input' },
+      );
+      expect(selectorWrapper).toBeInTheDocument();
+      expect(selectorInput).toBeInTheDocument();
+
+      // Expect no new option to be selectable.
+      userEvent.click(selectorInput);
+      userEvent.type(selectorInput, 'a new option');
+      act(() => jest.advanceTimersByTime(300));
+
+      expect(
+        container.querySelector('[role="option"]'),
+      ).not.toBeInTheDocument();
+      expect(within(selectorWrapper).getByText('No Data')).toBeInTheDocument();
     });
 
     it('renders with tokenSeparators', () => {
-      wrapper.setProps({ tokenSeparators: ['\n', '\t', ';'] });
-      expect(wrapper.find(SelectComponent)).toExist();
-      expect(wrapper.find(SelectComponent).prop('tokenSeparators')).toEqual(
-        expect.arrayContaining([expect.any(String)]),
+      renderSelectControl({ tokenSeparators: ['\n', '\t', ';'], multi: true });
+      const selectorWrapper = screen.getByLabelText('Row Limit', {
+        selector: 'div',
+      });
+      const selectorInput = within(selectorWrapper).getByLabelText(
+        'Row Limit',
+        { selector: 'input' },
       );
+      expect(selectorWrapper).toBeInTheDocument();
+      expect(selectorInput).toBeInTheDocument();
+
+      userEvent.click(selectorInput);
+      const paste = createEvent.paste(selectorInput, {
+        clipboardData: {
+          getData: () => '1 year ago;1 week ago',
+        },
+      });
+      fireEvent(selectorInput, paste);
+      const yearOption = screen.getByLabelText('1 year ago');
+      expect(yearOption).toBeInTheDocument();
+      expect(yearOption).toHaveAttribute('aria-selected', 'true');
+      const weekOption = screen.getByText(/1 week ago/, {
+        selector: 'div',
+      }).parentNode;
+      expect(weekOption?.getAttribute('aria-selected')).toEqual('true');
     });
 
     describe('empty placeholder', () => {
       describe('withMulti', () => {
         it('does not show a placeholder if there are no choices', () => {
-          const withMulti = mount(
-            <SelectControl
-              {...defaultProps}
-              choices={[]}
-              multi
-              placeholder="add something"
-            />,
-          );
-          expect(withMulti.html()).not.toContain('option(s');
+          const container = renderSelectControl({
+            choices: [],
+            multi: true,
+            placeholder: 'add something',
+          });
+          expect(
+            container.querySelector('[role="option"]'),
```

Comment:
queryByRole is better for this case.
```suggestion
            queryByRole('option')
```